src/shared/tools.cpp

```diff
@@ -95,7 +95,7 @@ int getuint(ucharbuf &p)
         n += (p.get() << 7) - 0x80;
         if(n & (1<<14)) n += (p.get() << 14) - (1<<14);
         if(n & (1<<21)) n += (p.get() << 21) - (1<<21);
-        if(n & (1<<28)) n |= -1<<28;
+        if(n & (1<<28)) n |= 0xF0000000;
```

Comment:
I don't see the motivation behind this change.

Reply:
Just like the commit message says, I think this is to remove a warning about shifting a negative value. Another way to solve this that would keep the shift is: `-1u<<28` or `~0u<<28`.